From 2a1d92348a86c1110b633d757d8944a5258a6f24 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:23 -0400
Subject: [usb] xhci: refactor TRB_EVAL_CONTEXT case into function

Message-id: <1410887463-20674-72-git-send-email-dzickus@redhat.com>
Patchwork-id: 95270
O-Subject: [RHEL7 PATCH 071/591] xhci: refactor TRB_EVAL_CONTEXT case into function
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit 07948a8da6241984b8359830498855573d552d0d
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Sep 9 13:29:53 2013 +0300

    xhci: refactor TRB_EVAL_CONTEXT case into function

    The function that handles xHCI command completion is much too long and
    there is need to be broken up into individual functions for each command
    completion to improve code readablity.
    This patch refactors the code in TRB_EVAL_CONTEXT switch case in
    handle_cmd_completion() into a fuction named xhci_handle_cmd_eval_ctx().

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index 6fb2f26..481ee04 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -1407,6 +1407,18 @@ static void xhci_handle_cmd_disable_slot(struct xhci_hcd *xhci, int slot_id)
  xhci_free_virt_device(xhci, slot_id);
 }
 
+static void xhci_handle_cmd_eval_ctx(struct xhci_hcd *xhci, int slot_id,
+  struct xhci_event_cmd *event, u32 cmd_comp_code)
+{
+ struct xhci_virt_device *virt_dev;
+
+ virt_dev = xhci->devs[slot_id];
+ if (handle_cmd_in_cmd_wait_list(xhci, virt_dev, event))
+  return;
+ virt_dev->cmd_status = cmd_comp_code;
+ complete(&virt_dev->cmd_completion);
+}
+
 static void xhci_handle_cmd_addr_dev(struct xhci_hcd *xhci, int slot_id,
   u32 cmd_comp_code)
 {
@@ -1552,11 +1564,8 @@ bandwidth_change:
   complete(&xhci->devs[slot_id]->cmd_completion);
   break;
  case TRB_TYPE(TRB_EVAL_CONTEXT):
-  virt_dev = xhci->devs[slot_id];
-  if (handle_cmd_in_cmd_wait_list(xhci, virt_dev, event))
-   break;
-  xhci->devs[slot_id]->cmd_status = GET_COMP_CODE(le32_to_cpu(event->status));
-  complete(&xhci->devs[slot_id]->cmd_completion);
+  xhci_handle_cmd_eval_ctx(xhci, slot_id, event,
+    GET_COMP_CODE(le32_to_cpu(event->status)));
   break;
  case TRB_TYPE(TRB_ADDR_DEV):
   xhci_handle_cmd_addr_dev(xhci, slot_id,
-- 
1.7.1